app/javascript/js/controllers/action_controller.js

```diff
@@ -24,9 +26,13 @@ export default class extends Controller {
 
   get resourceIds() {
     try {
-      return JSON.parse(document.querySelector(`[data-selected-resources-name="${this.resourceName}"]`).dataset.selectedResources)
+      return JSON.parse(this.dataSet.selectedResources)
     } catch (error) {
       return []
     }
   }
+
+  get dataSet() {
```

Comment:
This is a good refactor but let's name it a bit better please. How about `selectionOptions`?

Reply:
changed and it looks better this way